### PR TITLE
Mimes can no longer remake a vow of silence after breaking it

### PR DESCRIPTION
**Pull request by E231kei500bandai**
## About The Pull Request
I think being able to have your mime powers after waiting just 5 minutes kind of defeats the point of not being able to talk as mime. It's supposed to be a tradeoff where you get cool powers in exchange for not being able to talk. I assumed it was permanent already, but [I just found out that it wasn't](https://tgstation13.org/phpBB/viewtopic.php?f=33&t=33526) so I decided to change it. 

Also adds logging to breaking the vow. It seemed strange that it wasn't logged, but maybe it's not important enough to be logged or it's already logged somehow? Let me know. 

Also fixes the finger gun icon being an error

## Why It's Good For The Game
Breaking vow of silence is now permanent and mimes have to live with their decisions. Breaking the vow has an actual consequence of permanently losing abilities instead of just 5 minutes without them. Makes things more challenging for traitor mimes as they can't get their advanced mime abilities back after talking. (Although if they didn't read both books they can read one of them to get the abilities back, since reading the books adds another vow of silence. Maybe that should be removed too?)
## Changelog
:cl:
balance: Mimes can no longer make a vow of silence after breaking it
admin: Mimes breaking vows is now logged
fix: Fixes icon for mime finger gun ability
/:cl:

### CODE REVIEW

**Comment by iwishforducks** [comment]
i accidentally clicked the break the vow button as mime once and didn't speak during my vow break. could we just make it so it's like a safety toggle and when you speak it breaks it or to just make it take 10 minutes to retake the oath?

**Comment by BlueMemesauce** [comment]
> i accidentally clicked the break the vow button as mime once and didn't speak during my vow break. could we just make it so it's like a safety toggle and when you speak it breaks it or to just make it take 10 minutes to retake the oath?

Good idea, I'll make it like the microbomb implant activation
The downside of this is that if a mime is frantically trying to call for help, it will be harder to do so

**Review by MrMelbert — CHANGES_REQUESTED** [review]
No opinion on the pr (well I have an opinion, I don't like it that much but not strongly) but could do it correctly

File: code/modules/spells/spell_types/self/mime_vow.dm [hunk 1]
```diff
@@ -13,23 +13,29 @@
...
-		ADD_TRAIT(cast_on, TRAIT_MIMING, "[type]")
+	if(popup)
+		return FALSE
+	popup = TRUE
+	var/response = tgui_alert(cast_on, "Are you sure you want to break your vow of silence? This will disable your mimery abilities!", "Break Vow of Silence Confirmation", list("Yes", "No"))
+	popup = FALSE
+	if(response != "Yes")
+		return FALSE
+	var/datum/action/cooldown/spell/vow_of_silence/vow = locate() in cast_on.actions
+	vow.Remove(cast_on)
```
[the remark this answers is not in the document]
Comment (BlueMemesauce) [reply code/modules/spells/spell_types/self/mime_vow.dm:37]: its possible they can regain the vow of silence again through the traitor book, so i think they should stay

File: code/modules/spells/spell_types/self/mime_vow.dm [hunk 2]
```diff
@@ -13,23 +13,29 @@
 
 	spell_max_level = 1
 
-/datum/action/cooldown/spell/vow_of_silence/Grant(mob/grant_to)
+	var/popup = FALSE // is the confirmation window open?
```
[the remark this answers is not in the document]
Comment (BlueMemesauce) [reply code/modules/spells/spell_types/self/mime_vow.dm]: didn't know about autodoc, but it doesnt do anything for this because it's only one line right?

**Comment by BlueMemesauce** [comment]
sorry

**Comment by GuillaumePrata** [comment]
Being allowed to remake your vow only once per shift would make it be a bigger deal without punishing new mimes that break it by mistake.

Sometimes a mime breaking their vow makes the round a lot more interesting, and if it is something they can't ever take back, it just won't happen anymore.

**Comment by VladinXXV** [comment]
Breaking your vow applies a -8 mood change moodlet until the vow is restored. For comparison, this is as bad as severe withdrawal, overdose, an SM Resonance Cascade, robbing a grave, losing your tail, or being extremely disgusted. If you cannot remake your vow, that just means you are recieving a permanent moodlet of this severity. The positive moodlet given to antags is worth only 4 mood change, so even a traitor mime is looking at permanent -4 mood change. This is extremely punishing.

This is on top of the already punishing loss of your mime abilities itself and, assuming you actually spoke in a way such that people can hear you, becoming valid to literally everyone.

Five minutes to remake the vow is already a very long time to deal with. Breaking the vow has _legitimate_ uses that fall within what I would consider "acceptable mime gameplay." For example, an antagonist mime could break their vow to create voice activated bombs to punish their targets for speaking particular phrases. Consider the tip "As the Mime, you can use :r and :l to speak through your ventriloquist dummy." This was absolutely the case at one point, but it has been broken for a very long time now. It has been reported as a bug three times (#64736, #49582, #40969) and all three were ultimately closed as "not a bug". This behavior is shared with speaking through severed heads. Using the dummy is very much a part of playing mime, and dedicated mimes will break their vow to use the dummy in spite of the existing 5 minutes of -8 moodlet.

I could accept increasing the time to re-vow as high as 10 minutes, but to make it a permanent choice is using a sledgehammer to swat a fly. You do more damage to the role of Mime and what it can be than you prevent.

**Pull request closed without merging**